### ISSUE TITLE
Bug in ModelTreeWalker

### ISSUE THREAD

**Issue by RobbeDP** [lblod/ember-rdfa-editor#144]
The `ModelTreeWalker` returns wrong results in some cases when an empty range is given. This is however not always the case. Below are two examples, one that works as expected and one that doesn't work as expected. The paths at some point in the XML representation are indicated by an array of numbers.

### Works as expected:
```xml
<root>
  <div>
    [0, 0]<text>inside</text>[0, 6]
  <div>
  [1, 0]<text>outside</text>[1, 7]
</root>
```

When instantiating as follows `new ModelTreeWalker<ModelNode>({filter: toFilterSkipFalse(ModelNode.isModelText), range: range})` with `range` equals to [0] - [0, 0], we get an empty list as result.

### Doesn't work as expected:
```xml
<root>
  [0]<text>outside</text>[7]
  <div>
    [7, 0]<text>inside</text>[7,6]
  </div>
  [8]
</root>
```

When instantiating as follows `new ModelTreeWalker<ModelNode>({filter: toFilterSkipFalse(ModelNode.isModelText), range: range})` with `range` equals to [7, 6] - [8], we get the following list as result:
- `<text>outside</text>`
- `<text>inside</text>`

This should however be and empty list.

**Comment by RobbeDP** [comment]
Should by fixed in #145.

**Comment by nvdk** [comment]
@elpoelma could you check if this is fixed?

**Comment by elpoelma** [comment]
I will look into it.

**Comment by elpoelma** [comment]
Solved in #282
